### PR TITLE
67019 Setup e2e tests for Toxic Exposure

### LINKED ISSUE #67019: Toxic Exposure: Create E2E Tests
## Issue Description
-Create a new cypress spec to run TE tests with toggle on (the existing all-claims tests have a lot going on and already take quite a while to run)
-Add mock data needed for new Toxic Exposure Pages. 
-Add any necessary helper utils
-Add a new test for when the toggle is off to verify toxic exposure page(s) do not load

## Acceptance Criteria
-Verify e2e tests are successfully running for all applicable flows - all claims, toxic exposure

### PR DESCRIPTION
**Pull request by christinec-fftc**
## Summary
- Create a new cypress spec to run TE tests with TE toggle enabled 

- Add new tests for when the toggle is disabled to verify unreleased pages, such as toxic exposure page(s), do not load

- Update TE pages to have common path

old TE url's
toxic-exposure-conditions
gulf-war-hazard-1990
gulf-war-1990-location-<location_name>, e.g. gulf-war-1990-afghanistan

new TE url's
toxic-exposure/conditions
toxic-exposure/gulf-war-hazard-1990
toxic-exposure/gulf-war-1990-location-<location_name>

## Related issue(s)

- _Link to ticket created in va.gov-team repo_
department-of-veterans-affairs/va.gov-team#67019

## Testing done
- Submit transformer unit tests updated for TE. e2e tests created for TE
- Manual verification: url's have been updated as noted above. This is the only user facing change.

## Screenshots
n/a

## What areas of the site does it impact?
526EZ

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/disability-benefits/all-claims/tests/fixtures/data/maximal-toxic-exposure-test.json [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+    },
+    "gulfWar1990": {
+      "airspace": true
+    },
+    "gulfWar1990Locations": {
+      "afghanistan": {
+        "startDate": "1990-01-01",
+        "endDate": "1990-12-31"
+      }
+    },
```
Comment (christinec-fftc) [inline src/applications/disability-benefits/all-claims/tests/fixtures/data/maximal-toxic-exposure-test.json:150]: This json fixture was copied from `src/applications/disability-benefits/all-claims/tests/fixtures/data/maximal-test.json` and then lines 139 - 150 added for toxic exposure

File: src/applications/disability-benefits/all-claims/tests/fixtures/data/transformed-data/maximal-toxic-exposure-test.json [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+    },
+    "gulfWar1990": {
+      "airspace": true
+    },
+    "gulfWar1990Locations": {
+      "afghanistan": {
+        "startDate": "1990-01-01",
+        "endDate": "1990-12-31"
+      }
+    },
```
Comment (christinec-fftc) [inline src/applications/disability-benefits/all-claims/tests/fixtures/data/transformed-data/maximal-toxic-exposure-test.json:47]: This file was copied from `src/applications/disability-benefits/all-claims/tests/fixtures/data/transformed-data/maximal-test.json` and lines 36 - 47 added for TE

File: src/applications/disability-benefits/all-claims/tests/toxic-exposure.cypress.spec.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+    useWebComponentFields: true,
+
+    // skip: [],
+  },
+  manifest,
+  formConfig,
+);
+
+testForm(testConfig);
```
Comment (christinec-fftc) [inline src/applications/disability-benefits/all-claims/tests/toxic-exposure.cypress.spec.js:44]: @aurora-a-k-a-lightning i was thinking about your point about adding a negative test. I looked through the existing all-claims e2e tests and it doesn't look like we have a test for CFI only. I imagine this is a common scenario we should cover and in that flow, TE pages _should not_ display. 

If we think this is a good idea, I can add it in. However, given how much time I've already spent on this story, I'm inclined to create a separate story for that. lmk wdyt 😄

**Review by Mottie — DISMISSED** [review]
I have a couple of nit-picks, otherwise it looks good!

File: src/applications/disability-benefits/all-claims/tests/fixtures/data/maximal-toxic-exposure-test.json (hunk 1 above)
Comment (christinec-fftc) [reply src/applications/disability-benefits/all-claims/tests/fixtures/data/maximal-toxic-exposure-test.json:152]: I can add another location that has another set of start and/or end dates, but i do also want to make sure we cover the scenario where a Veteran selects a location, but doesn't recall the dates (which is what's currently being done for `"airspace"` - the dates are optional).

At some point when we switch from `va-memorable-date` to `va-date`, we should also check the scenario for year only

Comment (christinec-fftc) [reply src/applications/disability-benefits/all-claims/tests/fixtures/data/maximal-toxic-exposure-test.json:157]: fyi added `iraq` with dates

File: src/applications/disability-benefits/all-claims/tests/toxic-exposure.cypress.spec.js (hunk 3 above)
Comment (aurora-a-k-a-lightning) [reply src/applications/disability-benefits/all-claims/tests/toxic-exposure.cypress.spec.js:44]: @christinec-fftc separate story sounds good! 👍

**Review by aurora-a-k-a-lightning — APPROVED** [review]
lgtm! 🚢

**Pull request merged**